homeassistant/components/openrgb/__init__.py

```diff
@@ -0,0 +1,213 @@
+                hass.data[DOMAIN]["pending"][ha_type] = dev_ids
+                hass.async_create_task(
+                    hass.config_entries.async_forward_entry_setup(entry, "light")
+                )
+                hass.data[DOMAIN][ENTRY_IS_SETUP].add(config_entries_key)
+            else:
+                async_dispatcher_send(
+                    hass, ORGB_DISCOVERY_NEW.format("light"), device_list
+                )
+
+    def _get_updated_devices():
+        try:
+            orgb.update()
+        except ConnectionError:
+            hass.data[DOMAIN]["connection_failed"]()
+            return None
+        return orgb.devices
+
+    await async_load_devices(_get_updated_devices())
+
+    async def async_poll_devices_update(event_time):
+        if not hass.data[DOMAIN]["online"]:
+            # try to reconnect
+            try:
+                hass.data[DOMAIN][ORGB_DATA].comms.start_connection()
+                hass.data[DOMAIN]["connection_recovered"]()
+            except ConnectionError:
+                hass.data[DOMAIN]["connection_failed"]()
+
+        device_list = await hass.async_add_executor_job(_get_updated_devices)
+
+        if device_list is None:
+            return
+
+        await async_load_devices(device_list)
+
+        newlist_ids = []
+        for device in device_list:
+            newlist_ids.append(orgb_entity_id(device))
+        for dev_id in list(hass.data[DOMAIN]["entities"]):
+            # Clean up stale devices, or alert them that new info is available.
+            if dev_id not in newlist_ids:
+                async_dispatcher_send(hass, SIGNAL_DELETE_ENTITY, dev_id)
+                hass.data[DOMAIN]["entities"].pop(dev_id)
+            else:
+                async_dispatcher_send(hass, SIGNAL_UPDATE_ENTITY, dev_id)
+
+    hass.data[DOMAIN][ORGB_TRACKER] = async_track_time_interval(
+        hass, async_poll_devices_update, TRACK_INTERVAL
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_PULL_DEVICES, async_poll_devices_update
+    )
+
+    async def async_force_update(call):
+        """Force all devices to pull data."""
+        async_dispatcher_send(hass, SIGNAL_UPDATE_ENTITY)
+
+    hass.services.async_register(DOMAIN, SERVICE_FORCE_UPDATE, async_force_update)
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
+    """Unloading the OpenRGB platforms."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(
+                    entry, component.split(".", 1)[0]
+                )
+                for component in hass.data[DOMAIN][ENTRY_IS_SETUP]
+            ]
+        )
+    )
+
+    if unload_ok:
+        hass.data[DOMAIN][ENTRY_IS_SETUP] = set()
+        hass.data[DOMAIN][ORGB_TRACKER]()
+        hass.data[DOMAIN][ORGB_TRACKER] = None
+        hass.data[DOMAIN][ORGB_DATA].comms.stop_connection()
+        hass.data[DOMAIN][ORGB_DATA] = None
```

Comment:
What is the purpose to unsetting all of these on data that will be popped on line 211?